src/Common/Commands.Common.Authentication.Test/ClientFactoryHandlerTests.cs

```diff
@@ -65,7 +69,69 @@ public void DelegatingHandlersAreCloned()
             client = factory.CreateClient<StorageManagementClient>(context, AzureEnvironment.Endpoint.ServiceManagement);
             client = factory.CreateClient<StorageManagementClient>(context, AzureEnvironment.Endpoint.ServiceManagement);
             client = factory.CreateClient<StorageManagementClient>(context, AzureEnvironment.Endpoint.ServiceManagement);
-            Assert.Equal(5, MockDelegatingHandler.cloneCount); 
+            Assert.Equal(5, MockDelegatingHandler.cloneCount);
+        }
+
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void AddsAppropriateRetryPolicy()
+        {
+            AzureSessionInitializer.InitializeAzureSession();
+            string userAccount = "user@contoso.com";
+            Guid subscriptionId = Guid.NewGuid();
+            var account = new AzureAccount()
+            {
+                Id = userAccount,
+                Type = AzureAccount.AccountType.User,
+            };
+            account.SetTenants("common");
+            var sub = new AzureSubscription()
+            {
+                Id = subscriptionId.ToString(),
+            };
+            sub.SetAccount(userAccount);
+            sub.SetEnvironment("AzureCloud");
+            sub.SetTenant("common");
+            AzureContext context = new AzureContext
+            (
+                sub,
+                account,
+                AzureEnvironment.PublicEnvironments["AzureCloud"]
+            );
+
+            AzureSession.Instance.AuthenticationFactory = new MockTokenAuthenticationFactory(userAccount, Guid.NewGuid().ToString());
+            var factory = new ClientFactory();
+            factory.AddHandler(new RetryTestHandler());
+            var client = factory.CreateClient<StorageManagementClient>(context, AzureEnvironment.Endpoint.ServiceManagement);
+            var hyakHandler = EnsureHyakRetryPolicy(client);
+            hyakHandler.MaxTries = 2;
+            Assert.Throws<InvalidOperationException>(() => client.StorageAccounts.List());
+            hyakHandler.MaxTries = 0;
+            Assert.Throws<TaskCanceledException>(() => client.StorageAccounts.List());
+            var autorestClient = factory.CreateArmClient<ResourceManagementClient>(context, AzureEnvironment.Endpoint.ResourceManager);
+            var autoRestHandler = EnsureAutoRestRetryPolicy(autorestClient);
+            autoRestHandler.MaxTries = 2;
+            var task = autorestClient.ResourceGroups.ListWithHttpMessagesAsync();
+            Assert.Throws<InvalidOperationException>(() => task.ConfigureAwait(false).GetAwaiter().GetResult());
+            autoRestHandler.MaxTries = 0;
+            task = autorestClient.ResourceGroups.ListWithHttpMessagesAsync();
+            Assert.Throws<TaskCanceledException>(() => task.ConfigureAwait(false).GetAwaiter().GetResult());
+
+
```

Comment:
@markcowl do we need an assert for a successful retry of a list call and validating that content was returned from the call?

Reply:
I think we have that covered elsewhere